rules/external_binary/def.bzl

```diff
@@ -12,22 +12,40 @@ def _external_binary_impl(ctx):
     else:
         fail("Unsupported operating system: {}".format(ctx.os.name))
 
+    url = info.get("url")
     args = {
-        "url": info.get("url"),
+        "url": url,
         "sha256": info.get("sha256", ""),
-        "output": "{name}/{name}".format(name = ctx.attr.name),
     }
-    if info.get("is_compressed", False):
-        ctx.download_and_extract(**args)
+    if any([url.endswith(suffix) for suffix in [".zip", ".tar.gz", ".tgz", ".tar.bz2", ".tar.xz"]]):
+        ctx.download_and_extract(output="{name}/{name}_out".format(name = ctx.attr.name), **args)
+        build_contents = """
+        package(default_visibility = ["//visibility:public"])
+        filegroup(
+            name = "{name}_filegroup",
+            srcs = glob(["**/{name}*"]),
+        )
+        genrule(
+            name = "{name}",
+            srcs = [":{name}_filegroup"],
+            outs = ["executable"],
+            cmd_bash = "cp $(location :{name}_filegroup) $@", # Runs on *nix.
+            cmd_bat = "copy $(location :{name}_filegroup) $@", # Runs on Windows.
+            executable = True,
+        )
```

Comment:
We could use [@bazel_skylib//rules:copy_file](https://github.com/bazelbuild/bazel-skylib/blob/1.0.2/docs/copy_file_doc.md) instead, probably.